### ISSUE TITLE
DNT ads not clicked, even when skipping is disabled

### ISSUE THREAD

**Issue by dhowe** [dhowe/AdNauseam#1560]
![image](https://user-images.githubusercontent.com/737638/65369240-1736c980-dc7d-11e9-936a-351d0eac3e28.png)

**Comment by cqx931** [comment]
The two screenshots don’t look like under the same setting to me. In the first screenshot, ad clicking is still disabled, while in the second one it is not.

**Comment by dhowe** [comment]
Recreate as follows:
1. Visit a DNT-listed site with an ad using default settings (ad should be visible)
2. In settings, unclick 'Don't hide non-tracking ads' under Hiding
3. Refresh webpage and notice the ad is no longer visible
4. In settings, unclick 'Don't click non-tracking ads' under Clicking
5. Refresh the page and see the case above

The problem is, I think, is that the ad is first detected when not-clicking DNT ads is enabled. But when this settings is changed and one returns to the page, the ad is not then clicked as expected. Thus you get the confusing behavior above.

**Comment by cqx931** [comment]
@dhowe I see. Under the current mechanism we only attempt to do ad clicking when the ad was first collected. If we were expecting the ad to be clicked after step 5, a re-evaluation is needed. Then it should not consider as a bug, but a change in the current method. 

It might be nice to change that but I don't see it as a necessary one?

**Comment by dhowe** [comment]
Agreed. But while I don't think it is critical, I do think it presents a potentially quite confusing experience for the user, where they have made a clear change to the settings that is apparently being ignored. 

I think the simple fix is that whenever we check if a new ad is a duplicate, we also check that its DNT settings match. If not, then it is not a duplicate.

For example, we visit page A and find ad X with 'dont-click-DNT' enabled (so it is not clicked). If we later visit page A and see ad X with 'dont-click-DNT' DISABLED, it is NOT marked as duplicate (even though we have seen it before) because the DNT setting is different. Instead it is treated as a new ad (and clicked as normal)

?

**Comment by cqx931** [comment]
I see... so you mean the dntAllowed attribute(True or False) is going to influence whether two same ads are going to be considered as duplicate or not?

**Comment by dhowe** [comment]
That is what I was thinking

**Comment by mneunomne** [comment]
From what I understood we would have to store in the ad data the clicking settings when it was collected

probably here:

https://github.com/dhowe/AdNauseam/blob/09aad2c0dac2e13642f9a7ffcb815407872aed3e/src/js/adn/core.js#L1624-L1635

and then here compare with the current clicking settings when we are checking weather it is a duplicate or not:

https://github.com/dhowe/AdNauseam/blob/09aad2c0dac2e13642f9a7ffcb815407872aed3e/src/js/adn/core.js#L1651-L1661

**Comment by mneunomne** [comment]
Just to clarify, the clicking settings to be stored are:

- Click DNT or not (only check this different if the ad is collect from DNT origin) 
- Click frequency (this only makes sense to consider not as duplicate if the new clicking frequency is greated than the previous one)
